src/modules/universal-discovery/components/finder/finder.component.js

```diff
@@ -65,6 +65,45 @@ export default class FinderComponent extends Component {
         return { locationsMap: this.locationsMap, activeLocations: this.activeLocations };
     }
 
+    /**
+     * Loads starting location data (if there is a need) and loads root locations
+     *
+     * @method handleRootLocations
+     * @memberof FinderComponent
+     */
+    handleRootLocations() {
+        // Starting location is already loaded or its default
+        if (this.startingLocation.id === this.props.startingLocationId) {
+            this.loadRootLocations();
+            return;
+        }
+
+        this.props.loadLocation(
+            {...this.props.restInfo, locationId: this.props.startingLocationId},
+            (response) => {
+                if (response.View.Result.searchHits.searchHit[0].value.Location) {
+                    this.startingLocation = response.View.Result.searchHits.searchHit[0].value.Location;
```

Comment:
Here we should not introduce a new variable but use the `activeLocations` state and add it to the state.